File: week_05/yumyum/tuple.js
function solution(s) {
  let answer = [];
  let has = {};
  let sets = s.split("},").map((set) =>
    set
      .replace(/{|}/g, "")
      .split(",")
      .map((el) => Number(el))
  );

  sets.sort((a, b) => a.length - b.length);
  sets.forEach((set) => {
    set.forEach((el) => {
      if (!has[el]) {
        has[el] = true;
        answer.push(el);
      }
    });
  });

  return answer;
}

File: week_05/comet/tuple.kt
class Solution {
    fun solution(s: String): IntArray {
        val answer = mutableListOf<Int>()
        
        val elements = s.substring(1, s.length - 2).split("},").map {
            it.substring(1, it.length).split(',').map { it.toInt() }.toMutableList()
        }.sortedBy { it.size }
        
        elements.forEachIndexed { index, element ->
            answer.forEach {
                val index = element.lastIndexOf(it)
                element.removeAt(index)
            }

            answer.add(element[0])
        }
        
        return answer.toIntArray()
    }
}